addons/web/static/src/views/view_dialogs/form_view_dialog.js

```diff
@@ -30,6 +30,9 @@ export class FormViewDialog extends Component {
             preventEdit: this.props.preventEdit,
             discardRecord: () => {
                 this.props.close();
+                if (this.props.onRecordDiscarded) {
+                    this.props.onRecordDiscarded();
+                }
```

Comment:
by symmetry with saveRecord, I'd do

```suggestion
             async discardRecord: () => {
                if (this.props.onRecordDiscarded) {
                    await this.props.onRecordDiscarded();
                }
                this.props.close();
            },
```